---
title: "Chr2 hotspot preliminary analyses"
author: "Frederick Boehm"
date: "6/16/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
hot_chr <- 2
run_num <- 9941
hot_mid <- 165.5
```

```{r, load_attie}
load("~/Box Sync/attie/keller2018-chr2-hotspot-chtc/data-to-ignore/Attie_DO378_eQTL_viewer_v1.Rdata")
ls()
library(tidyverse)
options(tibble.print_max = 50, tibble.print_min = 20)
```

We need to get the names of the nonlocal traits that map to Chromosome 2 hotspot. 

I previously documented the issue with the Attie data set's hotspot annotations. Namely, when that file was created, they permitted the "hotspot" field to be only a single value per transcript. Thus, if a transcript mapped to multiple hotspots, it would still have only a single value in this "hotspot" field. Because of this, I need to first redo the mediation analyses that Keller did. 

We want the ensembl ids for the traits that map to Chr `r hot_chr` hotspot. First, define the boundaries of the hotspot. The supplemental table in Keller et al. 2018 lists only the midpoint of the hotspot. The half-width is 2 Mb.

```{r}
hot_lower <- hot_mid - 2
hot_upper <- hot_mid + 2
```

Now, filter to get those traits that map to the hotspot. Note that we're likely to get local traits, too, in this step.

```{r}
dataset.islet.rnaseq$lod.peaks %>%
  filter(chrom == hot_chr, pos <= hot_upper, pos >= hot_lower) %>%
  left_join(y = dataset.islet.rnaseq$annots, by = c("annot.id" = "gene_id")) %>%
  filter(!duplicated(annot.id)) %>%
  filter(lod >= 7.18)
```

The above yields expression traits that map to the hotspot. The list includes both local and nonlocal traits. We'll create two distinct objects, one for locals and one for nonlocal traits.

```{r}
local_annot <- dataset.islet.rnaseq$lod.peaks %>%
  filter(chrom == hot_chr, pos <= hot_upper, pos >= hot_lower) %>%
  left_join(y = dataset.islet.rnaseq$annots, by = c("annot.id" = "gene_id")) %>%
  filter(!duplicated(annot.id)) %>%
  filter(lod >= 7.18) %>%
  mutate(local_tx = chr == hot_chr) %>%
  filter(local_tx)
```

```{r}
dim(local_annot)
```



```{r}
nonlocal_annot <- dataset.islet.rnaseq$lod.peaks %>%
  filter(chrom == hot_chr, pos <= hot_upper, pos >= hot_lower) %>%
  left_join(y = dataset.islet.rnaseq$annots, by = c("annot.id" = "gene_id")) %>%
  filter(!duplicated(annot.id)) %>%
  filter(lod >= 7.18) %>% #genomewide threshold
  mutate(local_tx = chr == hot_chr) %>%
  filter(!local_tx) %>% 
  filter(!is.na(hotspot))
```


```{r}
dim(nonlocal_annot)
```

We'll use only those nonlocal traits that Keller et al. annotated as mapping to a hotspot.

This gives us the nonlocal traits for the Chr `r hot_chr` hotspot, which is what Keller et al. reported.

We also want to look at the local traits for Chr `r hot_chr` hotspot. We want to identify a subset of those for two-dimensional scans. Let's first limit to those near the hotspot, ie, within 5Mb of either end.

```{r, make_local_annot2}
local_annot2 <- local_annot %>%
  filter(middle <= hot_upper + 5, middle >= hot_lower - 5) %>%
  arrange(desc(lod))
```


Also, I'm limiting consideration of local genes to those with middle within 5Mb of the hotspot. 

Now, I make the two expression trait matrices that I need for use with Condor.

```{r}
hot_local <- dataset.islet.rnaseq$expr[, colnames(dataset.islet.rnaseq$expr) %in% local_annot2$annot.id]
hot_nonlocal <- dataset.islet.rnaseq$expr[, colnames(dataset.islet.rnaseq$expr) %in% nonlocal_annot$annot.id]
```

Prepare the covariates matrix for input to `fit1_med`.

```{r define_addcovar}
addcovar <- dataset.islet.rnaseq$covar[rownames(dataset.islet.rnaseq$covar) %in% rownames(hot_local), 1:4] 
```

Calls to `fit1_med`

```{r install_intermediate}
devtools::install_github("fboehm/qtl2mediate")
```

We then define the driver object. Note that we use the marker that corresponds to each nonlocal trait's peak position.

```{r define_dri}
# need the index of the peak for every nonlocal trait
indices <- nonlocal_annot %>%
  mutate(marker_index = purrr::map_int(.x = marker.id, .f = function(x){which(names(map[[as.character(hot_chr)]]) == x)})) %>%
  select(marker_index) %>%
  unlist() %>%
  as.integer()
```

```{r}
nonlocal_annot$marker.id[1] -> aa
which(names(map[[as.character(hot_chr)]]) == aa)
```






```{r fit1_med}
nlocal <- ncol(hot_local)
nnonlocal <- ncol(hot_nonlocal)
out <- list()

for (j in 1:nlocal){
  foo <- list()
  for (i in 1:nnonlocal){
    foo[[i]] <- qtl2mediate:::fit1_med_qtl2(driver = genoprobs[[as.character(hot_chr)]][, , indices[i]],
                                             target = hot_nonlocal[ , i],
                                             mediator = hot_local[ , j],
                                             addcovar = addcovar
                                             )
  }
  out[[j]] <- foo
}
```

We need to be careful when creating the `local_gene_id` and `nonlocal_gene_id` objects. 

```{r tabulate, message=FALSE, warning = FALSE}
tib <- out %>%
  purrr::map(bind_rows) %>%
  bind_rows() %>%
  as_tibble() %>%
  rename(lod_without_med = lod_no_med, lod_with_med = lod_med) %>%
  mutate(lod_diff = lod_without_med - lod_with_med, lod_diff_proportion = lod_diff / lod_without_med) %>%
  mutate(local_gene_id = rep(colnames(hot_local), each = nnonlocal), nonlocal_gene_id = rep(colnames(hot_nonlocal), times = nlocal)) %>%
  left_join(local_annot2, by = c("local_gene_id" = "annot.id")) %>%
  left_join(nonlocal_annot, by = c("nonlocal_gene_id" = "annot.id")) %>%
  select(symbol.x, symbol.y, lod_without_med, lod_with_med, lod_diff, lod_diff_proportion, local_gene_id, nonlocal_gene_id) %>%
  rename(local_symbol = symbol.x, nonlocal_symbol = symbol.y)
```





We also added gene symbols to the tibble. 

## Reading pleiotropy test results

```{r read_pleio_results}
results_dir <- paste0("~/Box Sync/attie/keller-hotspots/results/pvl-run-", run_num)
dir(results_dir) -> fns
mylist <- list()
for (i in 1:length(fns)){
  mylist[[i]] <- read.table(file.path(results_dir, fns[i])) %>%
  as_tibble() %>%
  mutate(filename = fns[i]) %>%
  mutate(local_gene_id = stringr::str_split(fns[i], "_")[[1]][[3]], nonlocal_gene_id = stringr::str_split(stringr::str_split(fns[i], "_")[[1]][[4]], ".txt")[[1]][[1]]) %>%
  filter(lod == max(lod), trace == "profile1") %>%
  select(local_gene_id, nonlocal_gene_id, lod, filename)
}
```

## Scatter plots of LOD difference & pleiotropy test statistics

```{r plot_all}
p <- mylist %>%
  bind_rows() %>%
  rename(pleiotropy_lod = lod) %>%
  left_join(tib, by = c("local_gene_id", "nonlocal_gene_id")) %>%
  select(local_symbol, nonlocal_symbol, pleiotropy_lod, lod_diff, lod_diff_proportion, lod_without_med, lod_with_med, filename, nonlocal_gene_id, local_gene_id) %>%
  ggplot() + geom_point(mapping = aes(x = pleiotropy_lod, y = lod_diff, colour = local_symbol, nonlocal = nonlocal_symbol), size = 0.1)
plotly::ggplotly(p)

```



```{r, plotly_by_local, fig.height=ceiling(nlocal / 2)}
p <- mylist %>%
  bind_rows() %>%
  rename(pleiotropy_lod = lod) %>%
  left_join(tib, by = c("local_gene_id", "nonlocal_gene_id")) %>%
  select(local_symbol, nonlocal_symbol, pleiotropy_lod, lod_diff, lod_diff_proportion, lod_without_med, lod_with_med, filename, nonlocal_gene_id, local_gene_id) %>%
  ggplot() + geom_point(mapping = aes(x = pleiotropy_lod, y = lod_diff, colour = local_symbol, nonlocal = nonlocal_symbol), size = 0.1) + facet_wrap(. ~ local_symbol, nrow = ceiling(nlocal / 2), ncol = 2) + theme(legend.position = "none")
plotly::ggplotly(p)
```


```{r, plotly_by_nonlocal, fig.height=ceiling(nnonlocal / 4)}
p <- mylist %>%
  bind_rows() %>%
  rename(pleiotropy_lod = lod) %>%
  left_join(tib, by = c("local_gene_id", "nonlocal_gene_id")) %>%
  select(local_symbol, nonlocal_symbol, pleiotropy_lod, lod_diff, lod_diff_proportion, lod_without_med, lod_with_med, filename, nonlocal_gene_id, local_gene_id) %>%
  ggplot() + geom_point(mapping = aes(x = pleiotropy_lod, y = lod_diff, colour = local_symbol, nonlocal = nonlocal_symbol), size = 0.1) + facet_wrap(. ~ nonlocal_symbol, ncol = 4, nrow = ceiling(nnonlocal / 4)) + theme(legend.position = "none")
plotly::ggplotly(p)
```


## Filter to remove local traits that are outside hotspot

In initial analyses for each hotspot, I included too many local traits. Namely, I considered all local traits that are within 7 Mb of the hotspot midpoint. I wish to limit this to traits that are within 3 Mb of the hotspot midpoint.



```{r define_foo}
foo <- mylist %>%
  bind_rows() %>%
  rename(pleiotropy_lod = lod) %>%
  left_join(tib, by = c("local_gene_id", "nonlocal_gene_id")) %>%
  select(local_symbol, nonlocal_symbol, pleiotropy_lod, lod_diff, lod_diff_proportion, lod_without_med, lod_with_med, filename, nonlocal_gene_id, local_gene_id) 
```


```{r add_local_peak_positions}
foo %>%
  left_join(local_annot2, by = c("local_gene_id" = "annot.id")) %>%
  rename(local_peak_pos = pos, local_peak_lod = lod) %>%
  select(- marker.id, - chrom, - strand, - biotype, - local_tx) %>%
  filter(local_peak_pos < hot_upper, local_peak_pos > hot_lower)


```

```{r}
foo %>%
  left_join(local_annot2, by = c("local_gene_id" = "annot.id")) %>%
  rename(local_peak_pos = pos, local_peak_lod = lod) %>%
  select(- marker.id, - chrom, - strand, - biotype, - local_tx) %>%
  filter(local_peak_pos < hot_upper, local_peak_pos > hot_lower) %>%
  select(middle) %>% unlist() %>% range()

```

## Convert tibble to matrix for clustering of pleiotropy test statistics

```{r, hclust_pleio_stats}
bar <- foo %>%
  select(nonlocal_gene_id, local_gene_id, pleiotropy_lod) %>%
hotspots::tibble_to_matrix() %>%
  dist() %>%
  hclust() 
```

```{r load_colors}
library(RColorBrewer) 
hmcol <- colorRampPalette(brewer.pal(9, "GnBu"))(100)
```


```{r heat_pleio_stats}
foo %>%
  select(nonlocal_gene_id, local_gene_id, pleiotropy_lod, lod_diff) %>%
hotspots::tibble_to_matrix() %>%
  t() %>%
  gplots::heatmap.2(cexRow = 0.1, cexCol = 0.1, col = hmcol, trace = "none")
```

```{r heat_lod_diff}
foo %>%
  select(nonlocal_gene_id, local_gene_id, lod_diff) %>%
hotspots::tibble_to_matrix() %>%
  t() %>%
  gplots::heatmap.2(cexRow = 0.1, cexCol = 0.1, col = hmcol, trace = "none")
```

```{r heat_lod_diff0}
foo %>%
  select(nonlocal_gene_id, local_gene_id, lod_diff) %>%
  mutate(lod_diff = lod_diff * (lod_diff > 0)) %>%
hotspots::tibble_to_matrix() %>%
  t() %>%
  gplots::heatmap.2(cexRow = 0.1, cexCol = 0.1, col = hmcol, trace = "none")
```



```{r, define_hld, eval = FALSE}
hld <- foo %>%
  select(nonlocal_gene_id, local_gene_id, lod_diff)
plotly::plot_ly(as.data.frame(hld), z = as.data.frame(hld)$lod_diff, type = "heatmap")
```


## Compare dendograms 

Should we create a lod difference dendogram based on the mediations with a single local trait? Or with the full matrix of mediation analyses? 


```{r define_hc_objects}
hc_lod_diff <- foo %>%
  select(nonlocal_gene_id, local_gene_id, lod_diff) %>%
  hotspots::tibble_to_matrix() %>%
  dist() %>%
  hclust() 
hc_pleio <- foo %>%
  select(nonlocal_gene_id, local_gene_id, pleiotropy_lod) %>%
  hotspots::tibble_to_matrix() %>%
  dist() %>%
  hclust() 
```

```{r load_dendextend}
library(dendextend)
```

```{r dend_dist}
dend_diff(as.dendrogram(hc_pleio), as.dendrogram(hc_lod_diff))
```

## 2 x 2 tables

We binarize the continuum of mediation LOD difference values and that of pleiotropy test statistics. 

```{r 2x2_table}
mediation_threshold <- 1.5
mediation_threshold_proportion <- 0.5
pleiotropy_threshold <- 4
table(med = foo$lod_diff > mediation_threshold, pleio = foo$pleiotropy_lod <= pleiotropy_threshold)
```

```{r make_mylist}
med_vals <- 2 * (1:5)
pleio_vals <- 2 * (2:5)
mylist <- list()
for(i in 1:length(med_vals)){
  my_other_list <- list()
  for (j in 1:length(pleio_vals)){
    uniq_cts_nonlocal <- foo %>% 
      mutate(med = lod_diff > med_vals[i], pleio = pleiotropy_lod <= pleio_vals[j]) %>%
      group_by(med, pleio) %>%
      select(nonlocal_gene_id, med, pleio) %>%
      unique() %>%
      count() %>%
      rename(n_unique_nonlocal_gene_id = n)
    uniq_cts <- foo %>% 
      mutate(med = lod_diff > med_vals[i], pleio = pleiotropy_lod <= pleio_vals[j]) %>%
      group_by(med, pleio) %>%
      select(local_gene_id, med, pleio) %>%
      unique() %>%
      count() %>%
      rename(n_unique_local_gene_id = n) %>%
      left_join(uniq_cts_nonlocal, by = c("pleio", "med"))
    
    my_other_list[[j]] <- foo %>%
      mutate(med = lod_diff > med_vals[i], pleio = pleiotropy_lod <= pleio_vals[j]) %>%
      group_by(med, pleio) %>%
      count() %>%
      left_join(uniq_cts, by = c("med", "pleio")) %>%
      mutate(med_thr = med_vals[i], pleio_thr = pleio_vals[j])
  }
  mylist[[i]] <- my_other_list
} 
mylist
```

```{r}
mylist %>%
  map_df(bind_rows)
```

```{r}
mylist %>%
  map_df(bind_rows) %>%
  filter(as.logical(med), !as.logical(pleio))
```

```{r}
mylist %>%
  map_df(bind_rows) %>%
  filter(as.logical(med), as.logical(pleio))
```




```{r 2x2_table_proportion}
table(med = foo$lod_diff_proportion > mediation_threshold_proportion, pleio = foo$pleiotropy_lod <= pleiotropy_threshold)
```





```{r counts_lod_diff}
foo %>%
  filter(lod_diff > mediation_threshold, pleiotropy_lod <= pleiotropy_threshold) %>%
  group_by(local_symbol) %>%
  count() %>%
  arrange(desc(n))

```

```{r counts_lod_diff_prop_first}
foo %>%
  filter(lod_diff_proportion > mediation_threshold_proportion, pleiotropy_lod <= pleiotropy_threshold) %>%
  group_by(local_symbol) %>%
  count() %>%
  arrange(desc(n))

```


```{r counts_both}
foo %>%
  filter(lod_diff > mediation_threshold, lod_diff_proportion > mediation_threshold_proportion, pleiotropy_lod <= pleiotropy_threshold) %>%
  group_by(local_symbol) %>%
  count() %>%
  arrange(desc(n))
```


```{r counts_either}
foo %>%
  filter(lod_diff_proportion > mediation_threshold_proportion | lod_diff > mediation_threshold, pleiotropy_lod <= pleiotropy_threshold) %>%
  group_by(local_symbol) %>%
  count() %>%
  arrange(desc(n))

```


```{r counts_lod_diff_prop}
foo %>%
  filter(lod_diff_proportion > 0.2, pleiotropy_lod <= pleiotropy_threshold) %>%
  group_by(local_symbol) %>%
  count() %>%
  arrange(desc(n))

```



## More figures

Let's plot a matrix of pleiotropy test statistics for nonlocal - local pairs, ordered by local gene's chromosomal position (ie, coordinate of the middle of the gene per ENSEMBL). We could, for example, color cells by pleiotropy test statistic value. Maybe truncate for those over 10. 

## Still more figures: graph theory package

Can we represent, say, the collection of nonlocal traits in an undirected graph theory graph? We might say that nodes i and j are connected by an edge if they are both pleiotropic in pairwise tests with a single local trait.

```{r load_hotspots}
library(hotspots)
```

```{r}
mymat <- foo %>%
  tibble_to_matrix() %>%
  t() %>%
  (function(mat) mat <= 4) %>%
  make_adjacency_matrix()
myg <- mymat %>%
  igraph::graph_from_adjacency_matrix(mode = "undirected", weighted = TRUE, diag = FALSE) 
comm2 <- igraph::fastgreedy.community(myg)
```

### Find communities

```{r}
library(igraph)
```

```{r}
comm <- foo %>%
  tibble_to_matrix() %>%
  t() %>%
  make_adjacency_matrix() %>%
  igraph::graph_from_adjacency_matrix(mode = "undirected", weighted = TRUE, diag = FALSE) %>%
  fastgreedy.community()

```



