---
title: "Expression QTL hotspot dissection with mediation analysis and pleiotropy testing"
author: "Frederick J. Boehm, Brian S. Yandell, and Karl W. Broman"
date: "August 1, 2019"
bibliography: research.bib
linkReferences: true
link-citations: true 
output: 
  bookdown::html_document2:
    code_folding: hide
---

```{r child="intro.Rmd"}
```


```{r load-broman}
library(broman)
library(here)
```


```{r child="methods.Rmd"}
```






# Results

Simple hotspots: Chr 2, Chr 13

Complex hotspots: Chr 5

Undetermined (right now): Chr 7, Chr 11



We generated scatter plots of pleiotropy test statistics vs. mediation LOD difference values for each of the five hotspots (Figures).
Each point represents a pairing of a local trait with a nonlocal trait.
The mediation LOD difference value is that observed when the local trait is considered
as a candidate mediator of the QTL-nonlocal trait association.

In all five scatter plots, we observe few
points in the upper right quadrant of the figure.
That is, few pairs have high pleiotropy test statistics and
high mediation LOD difference values.

```{r setup}
library(knitr)
library(qtl2hotspots)
pleio_threshold <- 4
ult <- 15 # univariate lod threshold
```

```{r load_tidyverse}
library(tidyverse)
options(tibble.print_max = 50, tibble.print_min = 20)
```


```{r load_heatmaply}
library(heatmaply)
```

```{r saveRDS}
if (!file.exists(here("analysis", "data", "derived_data", "expr.rds"))){
  download.file("https://datadryad.org/bitstream/handle/10255/dryad.166774/Attie_DO378_eQTL_viewer_v1.Rdata?sequence=2",
              destfile = here("analysis", "data", "raw_data", "Attie_DO378_eQTL_viewer_v1.Rdata")
              )
  load(here("analysis", "data", "raw_data", "Attie_DO378_eQTL_viewer_v1.Rdata"))
  saveRDS(dataset.islet.rnaseq$lod.peaks, here("analysis", "data", "derived_data", "lod_peaks.rds"))
  saveRDS(dataset.islet.rnaseq$annots, here("analysis", "data", "derived_data", "annots.rds"))
  saveRDS(dataset.islet.rnaseq$expr, here("analysis", "data", "derived_data", "expr.rds"))
  saveRDS(K, here("analysis", "data", "derived_data", "kinship.rds"))
  saveRDS(map, here("analysis", "data", "derived_data", "map.rds"))
  saveRDS(genoprobs, here("analysis", "data", "derived_data", "genoprobs.rds"))
  saveRDS(dataset.islet.rnaseq$covar, here("analysis", "data", "derived_data", "covar.rds"))
}
```

```{r readRDS}
lod_peaks <- readRDS(here("analysis", "data", "derived_data", "lod_peaks.rds"))
annots <- readRDS(here("analysis", "data", "derived_data", "annots.rds"))
expr <- readRDS(here("analysis", "data", "derived_data", "expr.rds"))
```

```{r, load_purrr}
library(purrr)
```

```{r setup_continued}
hot_chr <- c(2, 5, 7, 11, 13)
hot_mid <- c(165.5, 146, 46, 71, 112.5)
keller_mediator <- c("Hnf4a", "Pdx1", "Fam83e", "Sat2", "Il6st")
inputs <- tibble(hot_chr, hot_mid, keller_mediator)
out <- pmap_dfc(inputs, .f = function(hot_chr, hot_mid, keller_mediator) knitr::knit_expand(text = readLines(here::here("analysis", "paper", "hotspot-one-chromosome.Rmd"))))
# note that ... argument must be a list, ie, it must be a single argument, not multiple args
```





## Chr 2 hotspot results

`r knitr::knit(text = out[[1]])`


### Chr 2 scatter plots
 
The scatter plot of pleiotropy test statistics and mediation LOD differences reveals that few local-nonlocal 
trait pairs have both a high value of the pleiotropy test statistic and a high value of the mediation LOD 
difference. Many points have low values of both, while some points demonstrate a high value of only mediation 
LOD difference or pleiotropy test statistic. 

Many points have negative values of mediation LOD difference. These correspond to pairs that, when conditioning 
on the putative mediator, the LOD for the nonlocal trait increased. 

The points are color-coded to identify those pairs that involve the local trait *Hnf4a*, which 
@keller2018genetic identified as a key mediator of traits at this hotspot. Blue points involve *Hnf4a*, while 
red points do not. We see that many blue point have large mediation LOD differences and small pleiotropy test 
statistics. In fact, most of the points with mediation LOD difference above 15 involve *Hnf4a* and have small 
pleiotropy test statistics. 

An unanticipated observation is that some points have mediation LOD differences of 


### Chr 2 heatmap




### Chr 2 index v position plot







## Chr 5 results


### Chr 5 scatter plots


### Chr 5 heatmap

The Chr 5 heatmap of pleiotropy test statistics 
reveals a block-diagonal pattern of blue coloring among the nonlocal traits. We observe the 
set of nonlocal traits clustered into two blocks. Within each block, any two traits share a 
QTL, ie, are pleiotropic. Yet, the off-diagonal cells, ie, those nonlocal trait pairs that 
include one trait from each of the two blocks, tend to have greater pleiotropy test 
statistics. Still, not all off-diagonal blocks have large pleiotropy test statistics. The 
mostly dark blue matrix on the diagonal (from column 22 to 99) features no cells with pleiotropy test statistics above 3.2. While we didn't determine p-values for these analyses, prior analyses suggest that pleiotropy test statistics below 3.2 are consistent with p-values above 0.05, and, thus, are consistent with a single pleiotropic QTL.

Columns 100 to 119 (and rows 100 to 119) are a second (symmetric) block with only small 
pleiotropy test statistics. Columns 120 to 202 contains cells for the third (symmetric) block. Column 203, the last column, contains a trait *Gm* that, based on pleiotropy test statistics, doesn't cluster near other traits.


### Chr 5 index v position plot







## Chr 7 results

### Chr 7 scatter plots


### Chr 7 heatmap


### Chr 7 index v position plot



## Chr 11 results

### Chr 11 scatter plots


### Chr 11 heatmap


### Chr 11 index v position plot



## Chr 13 results

### Chr 13 scatter plots


### Chr 13 heatmap


### Chr 13 index v position plot


Our analysis of the Chromosome 13 hotspot presents results much like those for the Chromosome 2 
hotspot, with the Chromosome 13 local transcript *Il6st* playing a role like that of *Hnf4a* in the 
Chr 2 hotspot. In examining the scatter plot (Figure ) of mediation LOD differences against 
pleiotropy test statistics for all local-nonlocal pairs, we see, as in Figure for the Chr 2 
hotspot, a single local trait *Il6st* has strong mediation effects on a collection of nonlocal traits. *Il6st*, first identified by @keller2018genetic, accounts for most of the nonlocal traits that are affected by the hotspot. 

The heatmap for pleiotropy statistics for pairwise tests involving nonlocal traits resembles the 
heatmap for the Chr2 hotspot. Nearly all nonlocal traits have small test statistics when paired 
with each other. This is visualized as a dark blue, symmetric square along the heatmap's diagonal 
with only a few outlier nonlocal traits that demonstrate distinct patterns in pleiotropy test 
statistics.













# Discussion

Pleiotropy tests complement mediation analyses in the following manner. When considering a
collection of potential mediators for a nonlocal trait, mediation analyses clearly identify the
mediator when there is, in fact, a mediator among the candidates. However, when there is no
mediator, mediation analyses provide little information about the genetics of the complex trait.
Pleiotropy testing is particularly useful in this setting, since a pleiotropy test always gives
an answer about the number of underlying QTL. A drawback of pleiotropy testing is that the scientific question that it addresses - do two traits share a single QTL - informs, but doesn't fully resolve, the genetics of the nonlocal complex traits under study.

- lod difference threshold for declaration of "mediation". 
  - esp with regards to Chr 5 hotspot, where the keller mediator has a lot of traits with lod differences under 3, but above the 1.5 threshold.
  
  
- look at outputs of `lod_int()`, ie, the support intervals for all traits in a hotspot. Could plot interval against cluster order, for example.








# Colophon

This report was generated on ` r Sys.time()` using the following computational environment and dependencies:

```{r colophon, cache = FALSE, echo = TRUE, tidy.opts=list(width.cutoff=50), size = "tiny", results = "markup"}
# which R packages and versions?
devtools::session_info()
```


The current Git commit details are:

```{r git2r, cache = FALSE, echo = TRUE}
# what commit is this file at?
git2r::repository(here::here())
```




