# Methods

@keller2018genetic identified five expression trait hotspots in their
study of pancreatic islet gene expression in 378 Diversity Outbred mice. They
defined a hotspot to be a 4-Mb region that affects at least 100 nonlocal expression
traits.

For each hotspot, we identified a set of local expression traits and a
set of nonlocal expression traits. All traits demonstrated LOD peaks that met
the genome-wide significance threshold, 7.18 [@keller2018genetic].
The nonlocal expression traits arose from genes
on other chromosomes, while the local traits reflected expression levels for
genes on the same chromosome as the hotspot and within 4 Mb of the hotspot
center. We ignored transcripts for genes on the hotspot chromosome but not within 4 Mb of
the hotspot center.


## Pleiotropy testing

### Two-dimensional, two-QTL scans


$$
vec(Y) = X vec(B) + vec(G) + vec(E)
$${#eq:mvlmm}

For each hotspot, we tested all pairs of traits, regardless of whether the traits were local or nonlocal. We performed a
two-dimensional, two-QTL scan over a genomic region that included the entirety of the
hotspot. We fitted bivariate linear mixed effects models (Equations [-@eq:mvlmm;-@eq:g;-@eq:e]) at each ordered pair of markers in a two-dimensional grid [@boehm2019testing].




$$
G \sim MN_{n x 2}(0, K, V_g)
$${#eq:g}

$$
E \sim MN_{n x 2}(0, I_n, V_e)
$${#eq:e}

and $G$ and $E$ are independent.

$$
\hat B = (X^T\hat \Sigma^{-1}X)^{-1}X^T\hat\Sigma^{-1}Y
$$

$$
\hat \Sigma = \hat V_g \otimes \hat K + \hat V_e \otimes I_n
$$


$$
l_{10}(\hat B, \hat \Sigma) = - \frac{1}{2}\log_{10}\left((Y - X\hat B)^T\hat \Sigma^{-1}(Y - X\hat B)\right) - n\log_{10}(2\pi) - \log_{10}|\hat \Sigma|
$$


For $n$ mice, $\Sigma$ is a $2n$ by $2n$ covariance matrix, while $V_g$ and $V_e$ are $2$ by $2$ covariance matrices. Estimates $\hat V_g$ and $\hat V_e$ are obtained via an expectation-maximization algorithm [@zhou2014efficient;@gemma2].


### Calculating likelihood ratio test statistics

From the two-dimensional scans, we calculated pleiotropy likelihood ratio test statistics for each
pair of traits [@jiang1995multiple;@boehm2019testing;@boehm2019thesis]. Because of the required
computing time for bootstrap analyses, we didn't obtain p-values for the pleiotropy tests. Instead,
we worked directly with the test statistic values.



$$
LOD = \log_{10} \left(\frac{\max_{\lambda_1, \lambda_2}L(B, \Sigma, \lambda_1, \lambda_2)}{\max_{\lambda}L(B, \Sigma, \lambda, \lambda)} \right)
$${#eq:lod}


$$
ll(\lambda_1, \lambda_2)
$$


## Mediation analysis

A mediation analysis in systems genetics begins with an identified QTL-expression trait association. In our case, we choose a QTL that affects a nonlocal trait.

For each hotspot and each local trait - nonlocal trait pair, we also performed a
mediation analysis to assess the extent which the local trait mediates the QTL effect on the nonlocal trait. Each mediation analysis involved fitting four linear models (Equations~\ref{model1}, \ref{model2}, \ref{model3}, \ref{model4}) [@chick2016defining].

$$
Y = \alpha 1 + WC + E
$${#eq:model1}

$$
Y = XB + WC + E
$${#eq:model2}






$$
Y = \alpha 1 + WC + M\beta + E
$${#eq:model3}

$$
Y = XB + WC + M\beta + E
$${#eq:model4}

In all of our mediation analyses, the nonlocal transcripts are the targets, while the local
transcripts serve as candidate mediators of the QTL-target relationship. We only considered as potential mediators those local traits with strong univariate LODs.
In Equations~\ref{model1} through \ref{model4}, $X$ is a matrix of founder allele dosages,
$B$ is a matrix
of founder allele effects, $W$ is a matrix of covariates, $C$ is a matrix of covariate
effects, $M$ is a vector of values for a single candidate mediator, $\beta$ is the effect
of the mediator on the complex trait, 1 is a vector with all entries set to 1, $\alpha$
is the mean trait value, and $E$ is a vector of normally distributed random errors. 

## Visualizations

We created scatter plots with the `ggplot2` package [@ggplot2] for the R
statistical computing environment [@r]. We initially plotted all local-nonlocal pairs for each hotspot. We highlighted in blue those points that correspond to pairs involving the putative mediator per [@keller2018genetic].

### Heatmaps

We created heatmaps to examine patterns among pleiotropy test results in each hotspot.
We represented the pleiotropy test statistics as a symmetric matrix, where each row was one expression
trait and each column was an expression trait. 
Excepting the cells on the diagonal, each cell, then, contained the pleiotropy test statistic
from the analysis involving the row's expression trait and the column's expression trait.
To create a heatmap, we first performed two hierarchical clusterings of rows. 
One hierarchical clustering involved only the rows and columns containing local traits, while the other contained only 
rows and columns containing nonlocal traits.
We calculated the Euclidean distances between all row pairs (in the respective submatrices) before clustering. We ordered the rows according to optimal leaf ordering [@hahsler2008getting;@dendextend]. 
We then combined the
local and nonlocal traits into a single heatmap for each hotspot. 
To achieve coloring contrasts in the heatmaps, we
set the color spectrum limits at 0 (dark blue) and 5 (yellow). Pleiotropy test statistics above 5, too, are 
represented by
yellow. We annotated rows by a collection of columns on the right-hand side of the heatmap. This binary matrix 
consists of columns that summarize mediation results. Each column represents a local trait. If a row 
corresponds to a nonlocal trait that is mediated by the local trait, then the cell is colored dark green; 
otherwise, the cell is light green. The last annotation column and the annotation row above the heatmap both 
designate local traits with dark blue and nonlocal traits with light blue.









