---
title: "Title Goes Here"
author:
  - Author One
  - Author Two
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: 
    bookdown::html_document2:
      fig_caption: yes
      reference_docx: "../templates/template.docx" # Insert path for the DOCX file
bibliography: references.bib
csl: "../templates/journal-of-archaeological-science.csl" # Insert path for the bib-style
abstract: |
  High-dimensional traits in multiparental populations of model organisms offer 
  unprecedented opportunities to understand relationships between genotypic 
  variation and phenotypic variation. The ability to infer founder haplotypes in 
  multiparental populations combined with the measurement of molecular phenotypes 
  in hundreds of mammalian subjects has produced insights into diverse biological 
  processes.
  
  Quantitative trait locus (QTL) mapping of one trait at a time provides insights 
  into genetic underpinnings of complex traits. Furthermore, multivariate QTL
  mapping enables identification of pleiotropic QTL. Mediation analysis is widely 
  used to identify causal relationships among pairs of biomolecular traits. We 
  examine and compare two strategies, mediation analysis and pleiotropy testing, 
  for understanding relationships among traits. We apply our approach to five 
  expression trait hotspots that Keller, et al. (2018) identified in pancreatic 
  islets from nearly 400 Diversity Outbred mice. 
  
  At each hotspot, we identify local transcripts and nonlocal transcripts. We then 
  perform bivariate pleiotropy tests for all pairs involving one local and one 
  nonlocal transcript. Our mediation analyses consider each local transcript as a 
  candidate mediator for each nonlocal transcript. We summarize our findings with 
  the pleiotropy test statistics and LOD difference statistics. We conclude with 
  recommended practices for dissection of expression trait hotspots.
keywords: |
  keyword 1; keyword 2; keyword 3
highlights: |
  These are the highlights. 
---


<!-- This is the format for text comments that will be ignored during renderings. Do not put R code in these comments because it will not be ignored. -->

```{r, setup, echo = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  warning = FALSE,
  message = FALSE,
  echo = FALSE,
  comment = "#>",
  fig.path = "../figures/"
)

library(hotspots) # Or use devtools::load_all('.', quiet = T) if your code is in script files, rather than as functions in the `/R` diretory
```

# Introduction

Here is a citation [@Marwick2017]

# Background

```{r demo, eval = FALSE}
1 + 1
```

# Methods

# Results

```{r}
install.packages("here")
```


```{r get-data, eval = FALSE}
# Note the path that we need to use to access our data files when rendering this document
my_data <- read.csv(here::here('analysis', 'data', 'raw_data', 'my_csv_file.csv'))
```

# Discussion

# Conclusion

# Acknowledgements

<!-- The following line inserts a page break when the output is MS Word. For page breaks in PDF, use \newpage on its own line.  -->
##### pagebreak

# References 
<!-- The following line ensures the references appear here for the MS Word or HTML output files, rather than right at the end of the document (this will not work for PDF files):  -->
<div id="refs"></div>

##### pagebreak

### Colophon

This report was generated on `r Sys.time()` using the following computational environment and dependencies: 

```{r colophon, cache = FALSE}
# which R packages and versions?
devtools::session_info()
```

The current Git commit details are:

```{r}
# what commit is this file at? 
git2r::repository(here::here())
```
