\documentclass[]{article}
\usepackage[colorinlistoftodos]{todonotes}
\usepackage[utf8]{inputenc}
\usepackage{lineno}
\linenumbers
\usepackage{setspace}
\usepackage{bbm}
\usepackage[left = 1in, right = 1in, top = 1.5in, bottom = 1in]{geometry}
%\usepackage[titletoc]{appendix}
\usepackage{amsmath}
\usepackage{hyperref}
\hypersetup{unicode=true,
%            pdftitle={Chapter 1},
%            pdfauthor={Frederick Boehm},
%            pdfborder={0 0 0},
            colorlinks=true,
            breaklinks=true}
\usepackage[yyyymmdd,hhmmss]{datetime}

\usepackage[
    backend=biber,
    style=chicago-authordate,
%    citestyle=authoryear,
    natbib=true, %citet and citep
    url=true,
    doi=true,
    eprint=true,
    maxbibnames=10,
    backref=true
]{biblatex}
\DeclareLanguageMapping{english}{english-apa}
\addbibresource{research.bib}
\usepackage{graphicx}
\usepackage{color, colortbl}
\usepackage{amssymb}
\usepackage{amsthm}
\usepackage{amsfonts}
%\usepackage{longtable}
%\usepackage[outdir=./figs/]{epstopdf}
%\usepackage{etoolbox} % for messing with page style & title pages
\usepackage{fancyhdr}
\setlength{\headheight}{15.2pt}
\pagestyle{fancyplain}
\fancyhf{}
\lhead{}
\chead{}
\rhead{ \fancyplain{}{\thepage} } % put page number in upper right corner
\rfoot{Compiled on \today\ at \currenttime}
\renewcommand{\headrulewidth}{0pt} % remove top line
\renewcommand{\footrulewidth}{0pt} % remove bottom line
\renewcommand{\headrule}{}%

\title{Expression QTL hotspot dissection with mediation analysis and pleiotropy testing}
\author{Frederick J. Boehm, Brian S. Yandell, and Karl W. Broman}
\date{\today}

\doublespacing

\begin{document}

\maketitle

<<setup, include=FALSE>>=
library(knitr)
opts_chunk$set(
concordance=TRUE,
echo = TRUE,
fig.lp = "fig:",
warning = FALSE,
message = FALSE)

options(tinytex.verbose = TRUE)
@


\listoftodos
\tableofcontents
\listoffigures
\listoftables
%\pagestyle{headings} % page numbers appear at bottom of page, centered!
%\markright{\hfill}
\section{Introduction}

Genetics studies in model organisms like mice can identify
genomic regions that affect complex traits, such as systolic blood pressure and body weight \citep{sax1923association,soller1976power,lander1989mapping,broman2009guide,jansen2007quantitative}. These genomic regions are called ``quantitative trait loci'' or ``QTL''.
A genome-wide QTL ``scan'' reveals
associations between genotypes and phenotypes by considering each position,
one at a time, as a candidate QTL for the trait of interest.
A region with strong evidence of association with a complex trait, then, defines a QTL (for that trait).
Because nearby markers have correlated
genotypes, a QTL in a two-parent cross often spans multiple megabases in length
and may contain more than a hundred genes.
Identification of the causal gene (for a given complex trait) from among those genes near the QTL is
challenging and may require costly and time-consuming experiments.
The growing need for greater QTL mapping resolution fueled development over the
last two decades of model organism multiparental populations for high-resolution QTL mapping \citep{de2017back,churchill2004collaborative,svenson2012high,huang2012multiparent,shivakumar2018soybean,huang2011analysis,kover2009multiparent,tisne2017identification,stanley2017genetic}.
With experimentalists now measuring tens of thousands of biomolecular traits
in multiparental populations, the systems genetics community needs multivariate
statistical tools to fully examine the large volumes of data \citep{keller2018genetic,chick2016defining}.
Identifying loci that affect multiple traits can aid in identifying biomolecular interactions and in clarifying complex trait genetic architecture. A QTL that affects more than one trait is called a ``pleiotropic'' QTL.
A test of pleiotropy vs. separate QTL is one multivariate statistical tool that will inform
complex trait genetics by enabling researchers to identify the number of
unique QTL in a genomic region of interest.


<<load-broman>>=
library(broman)
@


\section{Methods}

\citet{keller2018genetic} identified five expression trait hotspots in their
study of pancreatic islet gene expression in 378 Diversity Outbred mice. They
defined a hotspot to be a 4-Mb region that affects at least 100 expression
traits.

For each hotspot, we performed a series of mediation analyses and pleiotropy
tests. For each hotspot, we identified a set of local expression traits and a
set of nonlocal expression traits. All traits demonstrated LOD peaks that met
the genome-wide significance threshold. The nonlocal expression traits arose from genes
on other chromosomes, while the local traits reflected expression levels for
genes on the same chromosome as the hotspot and within 7 Mb of the hotspot
center.

\subsection{Pleiotropy testing}

For each hotspot, we paired each local trait with each nonlocal trait and performed a
two-dimensional, two-QTL scan over a genomic region that included the entirety of the
hotspot. We fitted bivariate linear mixed effects models at each ordered pair of markers \citep{boehm2019testing}.

\begin{equation}
vec(Y) = X vec(B) + vec(G) + vec(E)
\label{eq:mvlmm}
\end{equation}


From the two-dimensional scans, we calculated pleiotropy test statistics for each pair of traits \citep{jiang1995multiple,boehm2019testing}.

\begin{equation}
\Lambda = \max_{\lambda_1, \lambda_2}ll(\lambda_1, \lambda_2) - \max_{\lambda} ll(\lambda, \lambda)
\label{eq:lrt}
\end{equation}

\subsection{Mediation analysis}

For each hotspot and each local trait - nonlocal trait pair, we also performed a
mediation analysis. Each mediation analysis involved fitting four linear models (Equations~\ref{model1}, \ref{model2}, \ref{model3}, \ref{model4}).

\begin{equation}
Y = \alpha 1 + WC + E
\label{model1}
\end{equation}

\begin{equation}
Y = XB + WC + E
\label{model2}
\end{equation}

\begin{equation}
Y = \alpha 1 + WC + M\beta + E
\label{model3}
\end{equation}

\begin{equation}
Y = XB + WC + M\beta + E
\label{model4}
\end{equation}

In all mediation analyses, the nonlocal transcripts are the targets, while the local transcripts serve as candidate mediators of the QTL-target relationship.





\section{Results}




\section{Discussion}



\printbibliography


<<install_here>>=
install.packages("here", repos = "https://cloud.r-project.org")
@



\subsection{Colophon}

This report was generated on \Sexpr{Sys.time()} using the following computational environment and dependencies:

\begin{singlespacing}

<<colophon, cache = FALSE>>=
# which R packages and versions?
devtools::session_info()
@



The current Git commit details are:

<<git2r, cache = FALSE>>=
# what commit is this file at?
git2r::repository(here::here())
@

\end{singlespacing}




\end{document}
