---
title: "Founder allele effects for hotspot traits"
author: "Frederick Boehm"
date: "8/14/2019"
output: html_document
---




## Goal 

The goals here are to follow up on Karl's suggestion to examine contrasts among allele effects. Specifically, can we see patterns of founder alleles near each trait's univariate peak. 

The heatmap of allele effects itself doesn't really reveal patterns that are easily interpreted. Hence, the suggestion to examine contrasts. Specifically, for each diagonal block of traits, is there a single, informative contrast? (or, perhaps a set of contrasts for each diagonal block.)


## Load the needed objects

Here, we reuse code from ../Rmd/hotspots-paper.Rmd.

```{r presetup}
eval_code <- TRUE
knitr::opts_chunk$set(comment=NA, fig.width=6, fig.height=6, eval = eval_code, echo = TRUE)
```









```{r define-ult}
library(qtl2hotspots)

ult <- 15 
# univariate lod threshold
```

```{r load_tidyverse}
library(tidyverse)
options(tibble.print_max = 50, tibble.print_min = 20)
```



```{r readRDS}
lod_peaks <- readRDS(here::here("analysis", "data", "derived_data", "lod_peaks.rds"))
annots <- readRDS(here::here("analysis", "data", "derived_data", "annots.rds"))
expr <- readRDS(here::here("analysis", "data", "derived_data", "expr.rds"))
probs <- readRDS(here::here("analysis", "data", "derived_data", "genoprobs.rds"))
K <- readRDS(here::here("analysis", "data", "derived_data", "kinship.rds"))
covar <- readRDS(here::here("analysis", "data", "derived_data", "covar.rds"))
map <- readRDS(here::here("analysis", "data", "derived_data", "map.rds"))
```


```{r setup_continued}
hot_chr <- c(2, 5, 7, 11, 13)
hot_mid <- c(165.5, 146, 46, 71, 112.5)
keller_mediator <- c("Hnf4a", "Pdx1", "Fam83e", "Sat2", "Il6st")
inputs <- tibble(hot_chr, hot_mid, keller_mediator)
out <- pmap_dfc(inputs, .f = function(hot_chr, hot_mid, keller_mediator) knitr::knit_expand(text = readLines(here::here("analysis", "paper", "Rmd", "hotspot-one-chromosome.Rmd"))))
# note that ... argument for knit_expand must be a list, ie, it must be a single argument, not multiple args
```


`r if (eval_code) knitr::knit(text = out[[2]])`












```{r correlations}
local_nonlocal_annot %>%
  ungroup() %>%
# 
  select(18:25) %>% 
  as.matrix() %>%
  apply(FUN = as.numeric, MARGIN = 2) %>%
  (function(x){rownames(x) <- local_nonlocal_annot$symbol; return(x)}) %>%
  (function(x) x[ord2, ]) %>%
  t() %>%
  cor()
which(rn == "Igfbp3")
```

```{r BC-v-FG}
local_nonlocal_annot %>%
  ungroup() %>%
  (function(x) x[ord2, ]) %>%
  dplyr::select(symbol, 18:25) %>%
  mutate(contrast_BC_FG = abs((cB + cC) - (cF + cG)), row_num = 1:203) %>%
  ggplot() + geom_point(aes(y = contrast_BC_FG, x = row_num))
```

```{r BC-v-G}
p <- local_nonlocal_annot %>%
  ungroup() %>%
  (function(x) x[ord2, ]) %>%
  dplyr::select(symbol, 18:25) %>%
  mutate(contrast = abs((cB + cC) / 2 - cG), row_num = 1:203) %>%
  ggplot() + geom_point(aes(y = contrast, x = row_num)) + geom_smooth(aes(y = contrast, x = row_num))
plotly::ggplotly(p)
```

```{r BCGH:ADEF}
p <- local_nonlocal_annot %>%
  ungroup() %>%
  select(symbol, cA:cH) %>%
  (function(x) x[ord2, ]) %>%
  mutate(contrast = abs((cB + cC + cG + cH) - (cA + cD + cE + cF)), row_num = 1:203) %>%
  ggplot() + geom_point(aes(y = contrast, x = row_num)) + geom_smooth(aes(y = contrast, x = row_num))
plotly::ggplotly(p)

  
```

```{r BCD:F}
p <- local_nonlocal_annot %>%
  ungroup() %>%
  select(symbol, cA:cH, local_tx) %>%
  (function(x) x[ord2, ]) %>%
  mutate(contrast = abs((cB + cC + cD) / 3 - cF), row_num = 1:203) %>%
  ggplot() + geom_point(aes(y = contrast, x = row_num, colour = local_tx, symbol = symbol)) + geom_smooth(aes(y = contrast, x = row_num))
plotly::ggplotly(p)
```

```{r BCGH:ADEF}
med_small <- med_tib %>% 
  filter(local_symbol == "Rpl21")
p <- local_nonlocal_annot %>%
  ungroup() %>%
  select(symbol, cA:cH, local_tx) %>%
  (function(x) x[ord2, ]) %>%
  inner_join(med_small, by = c("symbol" = "nonlocal_symbol")) %>%
  mutate(contrast = abs((cB + cC + cG + cH) - (cA + cD + cE + cF))) %>%
  ggplot() + geom_point(aes(y = contrast, x = lod_diff_proportion)) + geom_smooth(aes(y = contrast, x = lod_diff_proportion))
plotly::ggplotly(p)

  
```
Cxcl12 shows a strong contrast between B & G. 

```{r B:G}
med_small <- med_tib %>% 
  filter(local_symbol == "Rpl21")
p <- local_nonlocal_annot %>%
  ungroup() %>%
  select(symbol, cA:cH, local_tx) %>%
  (function(x) x[ord2, ]) %>%
  inner_join(med_small, by = c("symbol" = "nonlocal_symbol")) %>%
  mutate(contrast = abs(cB - cG)) %>%
  ggplot() + geom_point(aes(y = contrast, x = lod_diff_proportion)) + geom_smooth(aes(y = contrast, x = lod_diff_proportion))
plotly::ggplotly(p)

  
```

```{r B:G2}
p <- local_nonlocal_annot %>%
  ungroup() %>%
  select(symbol, cA:cH, local_tx) %>%
  (function(x) x[ord2, ]) %>%
  mutate(contrast = abs(cB - cG), row_num = 1:203) %>%
  ggplot() + geom_point(aes(y = contrast, x = row_num, colour = local_tx, symbol = symbol)) + geom_smooth(aes(y = contrast, x = row_num))
plotly::ggplotly(p)
```

## Calculate all pairwise contrasts for main effects

```{r}
contrasts <- local_nonlocal_annot %>%
  ungroup() %>%
  select(symbol, cA:cH, local_tx, lod) %>%
  (function(x) x[ord2, ]) %>%
  mutate(row_num = 1:203, 
         AB = abs(cA - cB),
         AC = abs(cA - cC),
         AD = abs(cA - cD),
         AE = abs(cA - cE),
         AF = abs(cA - cF),
         AG = abs(cA - cG),
         AH = abs(cA - cH),
         BC = abs(cB - cC),
         BD = abs(cB - cD),
         BE = abs(cB - cE),
         BF = abs(cB - cF),
         BG = abs(cB - cG),
         BH = abs(cB - cH),
         CD = abs(cC - cD),
         CE = abs(cC - cE),
         CF = abs(cC - cF),
         CG = abs(cC - cG),
         CH = abs(cC - cH),
         DE = abs(cD - cE),
         DF = abs(cD - cF),
         DG = abs(cD - cG),
         DH = abs(cD - cH),
         EF = abs(cE - cF),
         EG = abs(cE - cG),
         EH = abs(cE - cH),
         FG = abs(cF - cG),
         FH = abs(cF - cH),
         GH = abs(cG - cH)
  ) 
ctr <- contrasts %>% 
  tidyr::gather(contrast, value, AB:GH) %>%
  group_by(symbol) %>%
  filter(value == max(value)) %>%
  arrange(row_num) %>%
  ungroup() %>%
  select(contrast)
ctr %>%
  unlist() %>% 
  table()
ctr_runs <- ctr %>%
  unlist() %>% 
  rle()
con2 <- contrasts %>%
  mutate(BC_A = abs((cB + cC) / 2 - cA),
         BC_D = abs((cB + cC) / 2 - cD),
         BC_E = abs((cB + cC) / 2 - cE),
         BC_F = abs((cB + cC) / 2 - cF),
         BC_G = abs((cB + cC) / 2 - cG),
         BC_H = abs((cB + cC) / 2 - cH),
         BC_FG = abs((cB + cC) - (cF + cG))
         )
  con2 %>% 
    ggplot() + geom_point(aes(x = row_num, y = BC_E, colour = local_tx)) + geom_smooth(aes(x = row_num, y = BC_E))
```

```{r}
# which pairwise contrast has the greatest variability?
contrasts %>%
  select(local_tx, 13:40) %>% #select only the contrast column
  filter(!local_tx) %>%
  select(- local_tx) %>%
  purrr::map(.f = function(x)var(x)) %>% unlist()
```

```{r}
con2 %>% 
  filter(!local_tx) %>%
    ggplot() + geom_point(aes(x = row_num, y = BC_FG, colour = local_tx)) + geom_smooth(aes(x = row_num, y = BC_FG))
```

Now, add vertical lines that correspond to partitions between diagonal "blocks" in the pleiotropy statistics heatmap. 

The lower left block ends just before transcript "Elmo1". That is, "Elmo1" is the first component of the second block. What is the row_num value of the entry that precedes "Elmo1"?

```{r}
con2 %>%
  filter(symbol == "Elmo1")
# 68 is the row_num value for "Elmo1". Let's draw a vertical line at 67.5
con2 %>%
  filter(symbol == "Cep68") # row_num 129, so draw a vertical line at 129.5


```




```{r}
con2 %>% 
  filter(!local_tx) %>%
    ggplot() + geom_point(aes(x = row_num, y = BC_FG, colour = local_tx)) + geom_smooth(aes(x = row_num, y = BC_FG)) + geom_vline(aes(xintercept = 67.5)) + geom_vline(aes(xintercept = 129.5))
```

```{r}
con2 %>%
  filter(!local_tx) %>%
  mutate(small = row_num < 67.5, big = row_num > 129.5) %>%
  ggplot() + geom_histogram(aes(x = BC_FG)) + facet_wrap(~ big + small)
```


```{r}
p <- contrasts %>%
  select(1, row_num, local_tx, 13:40) %>%
  tidyr::gather(key = "contrast", value = "value", AB:GH) %>%
  filter(!local_tx) %>%
  ggplot() + geom_point(aes(x = row_num, y = value, colour = symbol)) + facet_wrap(contrast ~ .) + theme(legend.position = "none")
plotly::ggplotly(p)
```





```{r}
  con2 %>% 
    ggplot() + geom_point(aes(x = row_num, y = BC_G, colour = local_tx)) + geom_smooth(aes(x = row_num, y = BC_G))

```

```{r}
  con2 %>% 
    ggplot() + geom_point(aes(x = row_num, y = BC_F, colour = local_tx)) + geom_smooth(aes(x = row_num, y = BC_F))
```

```{r}
  con2 %>% 
    ggplot() + geom_point(aes(x = row_num, y = BC_H, colour = local_tx)) + geom_smooth(aes(x = row_num, y = BC_H))
```

```{r}
  con2 %>% 
    ggplot() + geom_point(aes(x = row_num, y = BC_D, colour = local_tx)) + geom_smooth(aes(x = row_num, y = BC_D))
```


```{r}
  con2 %>% 
    ggplot() + geom_point(aes(x = row_num, y = BC_A, colour = local_tx)) + geom_smooth(aes(x = row_num, y = BC_A))
```


## Calculate all partitions & contrasts

```{r}
pp <- expand.grid(1:3, 
                  1:3,
                  1:3, 
                  1:3,
                  1:3, 
                  1:3,
                  1:3, 
                  1:3
                  )
effects_mat <- local_nonlocal_annot %>%
  ungroup() %>%
  select(cA:cH) %>%
  as.matrix()

```

```{r}
pp_tib <- pp %>%
  as_tibble()
effects_tib <- effects_mat %>%
  as_tibble() 
  
ef <- pp_tib %>%
  apply(MARGIN = 1, 
        FUN = function(x){apply(X = effects_tib, 
                                MARGIN = 1, 
                                FUN = function(a, 
                                               b = x){
                                  qtl2hotspots::calc_contrast(grouping = b, 
                                                              effects = a)
                                  }
                                )}
        ) 

ef2 <- ef %>%  apply(FUN = function(x) x[!is.na(x)], MARGIN = 1) %>% 
  lapply(FUN = function(x) bind_rows(x)) %>%
  purrr::map2(local_nonlocal_annot$symbol, .f = function(x,y){x %>% mutate(symbol = y)}) %>%
  bind_rows() 
ef2 %>%
  group_by(contrast) %>%
  summarise(v = var(value)) %>%
  arrange(desc(v))
```

```{r}
rn_tib <- tibble::tibble(symbol = rn, row_number = 1:203)
ef2 %>%
  filter(contrast == "B_F") %>%
  inner_join(rn_tib) %>%
  inner_join(local_nonlocal_annot, by = "symbol") %>%
  filter(!local_tx) %>%
  ggplot() + geom_point(aes(x = row_number, y = value)) + geom_vline(aes(xintercept = 68.5))
```

## Defining the boundaries from the heatmap

I need to examine the heatmap to see how many diagonal "blocks" there are, and where the endpoints are, from 1:203, for every block.

First, note that 1:182 are all nonlocal traits. Now, Adamts15 through Cntnap2, inclusive, seems to define the first block.

The second block starts right above Cntnap2, ie, with Elmo1 and ends roughtly at Cep68.

Let's look at the first "block". 

```{r}
which(rn_tib$symbol == "Elmo1")
which(rn_tib$symbol == "Adamts15")
abs_diff <- function(x){abs(x[1] - x[2])}
notgroup1 <- rn_tib %>%
  mutate(group1_member = row_number <= 68) %>%
  inner_join(ef2) %>%
  group_by(contrast, group1_member) %>%
  summarise(variance = var(value), mean = mean(value)) %>%
  arrange(contrast) %>%
  ungroup() %>%
  group_by(contrast) %>%
  filter(!group1_member) %>%
  arrange(variance) %>% 
  rename(variance_notgroup1 = variance, mean_notgroup1 = mean)

rn_tib %>%
  mutate(group1_member = row_number <= 68) %>%
  inner_join(ef2) %>%
  group_by(contrast, group1_member) %>%
  summarise(variance = var(value), mean = mean(value)) %>%
  arrange(contrast) %>%
  ungroup() %>%
  group_by(contrast) %>%
  filter(group1_member) %>%
  arrange(variance) %>% 
  rename(variance_group1 = variance, mean_group1 = mean) %>%
  inner_join(notgroup1, by = "contrast") %>%
  dplyr::select(- group1_member.x, - group1_member.y)

```


```{r}
ef2 %>%
  filter(contrast == "B_G") %>%
  inner_join(rn_tib) %>%
  inner_join(local_nonlocal_annot, by = "symbol") %>%
  filter(!local_tx) %>%
  ggplot() + geom_point(aes(x = row_number, y = value)) + geom_vline(aes(xintercept = 68.5)) + geom_smooth(aes(x = row_number, y = value))

```


```{r}
rn_tib %>%
  mutate(group1_member = row_number <= 68) %>%
  inner_join(ef2) %>%
  group_by(contrast) %>%
  summarise(t_stat = t.test(value ~ group1_member)$statistic) %>%
  arrange(desc(t_stat))
```

```{r}
ef2 %>%
  filter(contrast == "AGH_CEF") %>%
  inner_join(rn_tib) %>%
  inner_join(local_nonlocal_annot, by = "symbol") %>%
  filter(!local_tx) %>%
  ggplot() + geom_point(aes(x = row_number, y = value)) + geom_vline(aes(xintercept = 68.5))

```




```{r, define-contrasts-5-1}
contr <- ef2 %>%
  filter(contrast == "AGH_CEF") %>%
  inner_join(rn_tib) %>%
  arrange(row_number)
```



```{r, order-annotations-5-1}
# all annots 
annots_ord <- list(lod_diffs_tibble, 
                   lod_diff_proportions_tibble, 
                   allele_effects_tibble,
                   peak_positions, 
                   local_tibble,
                   lod_tibble, 
                   contr
                    ) %>% 
  purrr::map(.f = function(df, rn){
    out <- df[match(x = rn, table = df$symbol), ]
    return(out)
  }, 
  rn = rn)
ann7 <- annots_ord[[7]] %>% # contrast
  dplyr::select(value) %>%
  as.matrix() %>%
  (function(x){rownames(x) <- annots_ord[[7]]$symbol; return(x)})
```

```{r}
iheatmapr::main_heatmap(mymat, show_colorbar = FALSE, layout = list(font = list(size = 4))) %>%
  iheatmapr::add_col_labels() %>%
  iheatmapr::add_col_annotation(ann5, show_colorbar = FALSE) %>%
  iheatmapr::add_main_heatmap(ann3, name = "signal3", size = 0.2, show_colorbar = FALSE) %>%
    iheatmapr::add_col_labels() %>%

  iheatmapr::add_main_heatmap(ann2, name = "signal2", size = 0.5, show_colorbar = FALSE) %>%
  iheatmapr::add_col_labels() %>%
  iheatmapr::add_main_heatmap(ann1, name = "signal1", size = 0.5, show_colorbar = FALSE) %>%
  iheatmapr::add_row_annotation(ann5, side = "left", show_colorbar = TRUE) %>%
  iheatmapr::add_col_labels() %>%
  iheatmapr::add_row_annotation(ann4, side = "left", show_colorbar = TRUE) %>%
  iheatmapr::add_row_annotation(ann6, side = "left", show_colorbar = TRUE) %>%
  iheatmapr::add_row_annotation(ann7, side = "left", show_colorbar = TRUE) %>%
  iheatmapr::add_row_labels()

```



# Other ideas

What results do we get when we try to "find the contrast" for other hotspots: 2, 7, 11, 13?

What if more than one contrast is "responsible" for a given block in a hotspot? Maybe max of two contrasts, for example? Or min of two contrasts?






