# Methods

We used the publicly available data from @keller2018genetic to demonstrate our approach for 
expression trait hotspot dissection. @keller2018genetic measured genome-wide gene expression 
levels with RNA sequencing of pancreatic islets in 378 
Diversity Outbred mice. From the RNA sequencing counts, they obtained normalized gene 
expression levels for 21,771 transcripts, which they shared publicly via DataDryad [@dryad]. 
@keller2018genetic also included founder allele probabilities at 69,005 markers across the 
genome, covariates, and leave-one-chromosome-out estimated kinship matrices [@broman2019rqtl2; @yang2014advantages].

@keller2018genetic performed univariate QTL mapping for all expression traits. They
defined an expression trait hotspot to be a 4-Mb region that affects at least 100 nonlocal 
expression traits.
They identified five expression trait hotspots in their
study of pancreatic islet gene expression in 378 Diversity Outbred mice . 

In analyzing the data from @keller2018genetic, we first established criteria for identifying local and nonlocal transcripts at each of the five hotspots. We focused on those local 
traits that strongly associated with QTL within the hotspot. We set the univariate LOD threshold for 
local traits at 15. We limited consideration to those local traits for which the middle of the gene lay 
within the hotspot interval. 
All nonlocal traits demonstrated univariate LOD peaks that met
the genome-wide significance threshold, 7.18 [@keller2018genetic]. The nonlocal expression traits arose 
from genes
on other chromosomes, while the local traits reflected expression levels for
genes on the same chromosome as the hotspot. We ignored transcripts for genes on the hotspot chromosome 
but outside the hotspot interval. 

For each hotspot, we performed both pairwise pleiotropy tests and linear regression-based 
mediation analyses. For every pair of traits at a hotspot, we calculated a pleiotropy test 
statistic[@boehm2019testing]. Due to the computational burden, we didn't determine bootstrap p-values for these 
tests. For every local trait - nonlocal trait pair, we also performed a linear 
regression-based mediation analysis [@chick2016defining,@keller2018genetic]. We describe in detail both procedures below.

## Pleiotropy testing



\begin{equation}
vec(Y) = X vec(B) + vec(G) + vec(E)
(\#eq:mvlmm)
\end{equation}

For each hotspot, we tested all pairs of traits, regardless of whether the traits were local or nonlocal. 
We performed a
two-dimensional, two-QTL scan over a genomic region that included the entirety of the
hotspot. We fitted bivariate linear mixed effects models (Equations \@ref(eq:mvlmm), \@ref(eq:g), and \@ref(eq:e)) at each ordered pair of markers in a two-dimensional grid [@boehm2019testing].




\begin{equation}
G \sim MN_{n x 2}(0, K, V_g)
(\#eq:g)
\end{equation}

\begin{equation}
E \sim MN_{n x 2}(0, I_n, V_e)
(\#eq:e)
\end{equation}

and $G$ and $E$ are independent.

\begin{equation}
\hat B = (X^T\hat \Sigma^{-1}X)^{-1}X^T\hat\Sigma^{-1}Y
(\#eq:Bhat)
\end{equation}

\begin{equation}
\hat \Sigma = \hat V_g \otimes \hat K + \hat V_e \otimes I_n
(\#eq:Sigmahat)
\end{equation}

\begin{equation}
l_{10}(\hat B, \hat \Sigma) = - \frac{1}{2}\log_{10}\left((Y - X\hat B)^T\hat \Sigma^{-1}(Y - X\hat B)\right) - n\log_{10}(2\pi) - \log_{10}|\hat \Sigma|
(\#eq:loglik)
\end{equation}


For $n$ mice, $\Sigma$ is a $2n$ by $2n$ covariance matrix, while $V_g$ and $V_e$ are $2$ by $2$ covariance matrices. Estimates $\hat V_g$ and $\hat V_e$ are obtained via an expectation-maximization algorithm [@zhou2014efficient;@gemma2].


From the two-dimensional scans, we calculated pleiotropy likelihood ratio test (LOD) statistics for each
pair of traits [@jiang1995multiple;@boehm2019testing;@boehm2019thesis]. Because of the required
computing time for bootstrap analyses, we didn't obtain p-values for the pleiotropy tests. Instead,
we worked directly with the test statistic values.



\begin{equation}
LOD = \log_{10} \left(\frac{\max_{\lambda_1, \lambda_2}L(B, \Sigma, \lambda_1, \lambda_2)}{\max_{\lambda}L(B, \Sigma, \lambda, \lambda)} \right)
(\#eq:lod)
\end{equation}




## Mediation analysis

A mediation analysis in systems genetics begins with an identified QTL-expression trait 
association [@chick2016defining]. In our case, we choose a QTL that affects a nonlocal expression trait, then ask if any of 
the nearby local expression traits lie in the causal pathway from QTL to nonlocal trait. 

A single mediation analysis involving one nonlocal trait and one local trait requires fitting four linear models (Equations \@ref(eq:model1), \@ref(eq:model2), \@ref(eq:model3), \@ref(eq:model4)) [@chick2016defining].

\begin{equation}
Y = \alpha 1 + WC + E
(\#eq:model1)
\end{equation}

\begin{equation}
Y = XB + WC + E
(\#eq:model2)
\end{equation}

\begin{equation}
Y = \alpha 1 + WC + M\beta + E
(\#eq:model3)
\end{equation}

\begin{equation}
Y = XB + WC + M\beta + E
(\#eq:model4)
\end{equation}

In all of our mediation analyses, the nonlocal transcripts are the targets, while the local
transcripts serve as candidate mediators of the QTL-target relationship. 
We only considered as potential mediators those local traits with strong univariate LODs.
In Equations~\ref{eq:model1} through \ref{eq:model4}, $X$ is a matrix of founder allele dosages,
$B$ is a matrix
of founder allele effects, $W$ is a matrix of covariates, $C$ is a matrix of covariate
effects, $M$ is a vector of values for a single candidate mediator (a local expression trait, in our case), $\beta$ is the effect
of the mediator on the complex trait (after adjusting for other covariates in the model), 1 is a vector with all entries set to 1, $\alpha$
is a parameter that corresponds to the mean trait value (after adjusting for other covariates in the model), and $E$ is a vector of normally distributed random errors. 

We included four covariates in our models: three wave indicators and sex. The wave indicators are binary vectors to account for wave membership. @keller2018genetic phenotyped the mice in four waves of nearly 100 mice per wave. 


## Visualizations

We created scatter plots with the `ggplot2` package [@ggplot2] for the R
statistical computing environment [@r]. We plotted all local-nonlocal pairs for each hotspot. We highlighted in blue those points that correspond to pairs involving the putative mediator that @keller2018genetic identified.



We created heatmaps to examine patterns among pleiotropy test results in each hotspot. Our analysis used both 
local and nonlocal traits.
We represented the pleiotropy test statistics as a symmetric matrix, where each row was one expression
trait and each column was an expression trait. 
Excepting the cells on the diagonal, each cell, then, contained the pleiotropy test statistic
from the analysis involving the row's expression trait and the column's expression trait.
To create a heatmap, we first performed two hierarchical clusterings of rows. 
One hierarchical clustering involved only the rows and columns containing local traits, while the other contained only 
rows and columns containing nonlocal traits.
We calculated the Euclidean distances between all row pairs (in the respective 
submatrices) before clustering. Hierarchical clustering was performed using the complete 
linkage method. Initially, each row is assigned to its own cluster. An iterative 
algorithm then combines the two most similar clusters at each step. Distances between 
clusters were computed with the Lance-Williams dissimilarity update formula for complete 
linkage [@lance1967general]. Lastly, we arranged the rows according to optimal leaf ordering [@hahsler2008getting;@dendextend]. 
We then combined the
local and nonlocal traits into a single heatmap for each hotspot. 
To achieve coloring contrasts in the heatmaps, we
set the color spectrum limits at 0 (dark blue) and 5 (yellow). Pleiotropy test statistics above 5, too, are 
represented by
yellow. We annotated rows by a collection of columns on the right-hand side of the heatmap. This binary matrix 
consists of columns that summarize mediation results. Each column represents a local trait. If a row 
corresponds to a nonlocal trait that is mediated by the local trait (*i.e.*, with mediation LOD 
difference greater than 1.5), then the cell is colored dark green; 
otherwise, the cell is light green. We didn't examine the possibility that a local trait might 
mediate the relationship between a QTL and another local trait; thus, all local trait rows are 
colored light green. The last annotation column and the annotation row above the heatmap both 
designate local traits with dark blue and nonlocal traits with light blue.









